packages/wallet/src/services/WalletAssetProvider.ts

```diff
@@ -0,0 +1,214 @@
+import { Asset, AssetProvider, Cardano, GetAssetArgs, GetAssetsArgs, HealthCheckResponse } from '@cardano-sdk/core';
+import { Assets } from '../types';
+import { Logger } from 'ts-log';
+import { Observable, firstValueFrom } from 'rxjs';
+import { isNotNil } from '@cardano-sdk/util';
+
+export interface AssetProviderContext {
+  assetProvider: AssetProvider;
+  assetInfo$: Observable<Assets>;
+  tx: Cardano.Tx;
+  logger: Logger;
+}
+
+const tryCip68Metadata = (
```

Comment:
There's also Cip68 FT metadata (333 label) which we're not handling. In fact, we don't even have utils to parse it yet, and it's not being projected. Let's be more explicit here.
```suggestion
const tryCip68NftMetadata = (
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1075/commits/80c61e293ca1968f366c665694957293a8d41bfe